pkg/kritis/cryptolib/verifier.go

```diff
@@ -70,23 +81,32 @@ func NewVerifier(imageDigest string, publicKeySet []PublicKey) (Verifier, error)
 // VerifyAttestation verifies an Attestation. See Verifier for more details.
 func (v *verifier) VerifyAttestation(att *Attestation) error {
 	var (
-		err     error
-		payload []byte
+		err       error
+		payload   []byte
+		publicKey PublicKey
 	)
 
 	// Extract the public key from `publicKeySet` whose ID matches the one in
 	// `att`.
-	// TODO: Replace no-op with correct implementation.
-	publicKey := v.PublicKeySet[0]
+	foundKey := false
```

Comment:
I think you can avoid this variable by checking `publicKey == nil`

Reply:
Removed. Not necessary because public keys are now indexed at construction time. Finding is now a simple map lookup.